ElevenLabs/Packages/com.rest.elevenlabs/Runtime/TextToSpeech/TextToSpeechEndpoint.cs

```diff
@@ -256,6 +201,283 @@ void StreamCallback(Response partialResponse)
+                    {
+                        throw new ArgumentException("Failed to parse clip id!");
+                    }
+
+                    string chunkText = Encoding.UTF8.GetString(partialResponse.Data);
+                    textBuffer.Append(chunkText);
+                    TryFlushAccumulatedTextLines();
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError(e);
+                }
+            }
+
+            void TryFlushAccumulatedTextLines()
+            {
+                // Process any complete lines
+                var text = textBuffer.ToString();
+                var lines = text.Split('\n');
+
+                // Keep the last potentially incomplete line
+                textBuffer.Clear();
+                textBuffer.Append(lines[^1]);
+
+                // Process all complete lines
+                for (int i = 0; i < lines.Length - 1; i++)
+                {
+                    HandleResponseJsonLine(lines[i]);
+                }
+
+                void HandleResponseJsonLine(string line)
+                {
+                    line = line.Trim();
+                    if (string.IsNullOrEmpty(line)) return;
+
+                    try
+                    {
+                        var responseJson = JObject.Parse(line);
+
+                        string audioBase64 = responseJson["audio_base64"].ToString();
+                        byte[] audioBytes = Convert.FromBase64String(audioBase64);
+
+                        var pcmData = PCMEncoder.Decode(audioBytes, PCMFormatSize.SixteenBit);
+                        var audioClip = AudioClip.Create($"{clipId}_{++part}", pcmData.Length, 1, frequency, false);
+
+                        if (!audioClip.SetData(pcmData, 0))
+                        {
+                            Debug.LogError("Failed to set PCM data to partial clip.");
+                            return;
+                        }
+
+                        TimestampedTranscriptCharacter[] timestampedTranscriptCharacters = ExtractTranscriptCharacters(responseJson);
+
+                        partialClipCallback.Invoke(new PartialTranscribedAudioClip(audioClip, timestampedTranscriptCharacters));
+
+                        accumulatedPCMData.AddRange(pcmData);
+                        accumulatedTranscriptData.AddRange(timestampedTranscriptCharacters);
+                    }
+                    catch (JsonReaderException e)
+                    {
+                        Debug.LogWarning($"Failed to parse line as JSON: {e.Message}");
+                    }
+                }
+            }
+        }
+
+        private static TimestampedTranscriptCharacter[] ExtractTranscriptCharacters(JObject responseJson)
+        {
+            TimestampedTranscriptCharacter[] timestampedTranscriptCharacters = Array.Empty<TimestampedTranscriptCharacter>();
+            var alignmentToken = responseJson.SelectToken("alignment");
```

Comment:
how many of these tokens are there?

Reply:
It timestamps each character, if thats' what you mean? It tends to only actually send timestamps every 10 or so JSON objects though.